### PR TITLE
create signup page

### PR DESCRIPTION
**Pull request by aayushbatra02**

### CODE REVIEW

File: src/components/SuccessModal.vue [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<template>
```
Comment (amrit56789) [inline src/components/SuccessModal.vue]: Change the file name

File: src/components/MiniLoader.vue [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<template>
```
Comment (amrit56789) [inline src/components/MiniLoader.vue]: MiniLoader?

File: src/components/ConfirmationModal.vue [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<template>
+  <div
+    class="absolute w-full h-full bg-black/50 flex justify-center items-center"
+  >
+    <div
+      class="bg-white px-[1.5rem] lg:px-[3rem] py-[2rem] lg:py-[4rem] rounded-lg flex justify-center items-center flex-col"
+    >
+      <h1 class="text-lg lg:text-2xl font-bold text-blue">{{ heading }}</h1>
+      <p class="mt-4 text-lg">{{ para }}</p>
```
Comment (yuvrajKaexpertise) [inline src/components/ConfirmationModal.vue]: what is para? @aayushbatra02

File: src/components/ConfirmationModal.vue [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      <button
+        @click="$emit('confirmButtonHandler')"
```
Comment (yuvrajKaexpertise) [inline src/components/ConfirmationModal.vue]: do you think, the handler in the function name is important? @aayushbatra02

File: src/stores/signupStore.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    roles: [],
+    loading: false,
+    error: null,
+    isUserSignedup: false,
+    user: null,
+  });
+
+  const getRoles = async () => {
+    try {
+      const res = await axios.get(`${baseUrl}role/list`);
```
Comment (yuvrajKaexpertise) [inline src/stores/signupStore.js]: define base url in axios interceptor and append from there @aayushbatra02

**Review by yuvrajKaexpertise — CHANGES_REQUESTED** [review]

File: src/components/ConfirmationModal.vue (hunk 4 above)
Comment (aayushbatra02) [reply src/components/ConfirmationModal.vue]: no sir i will change it

File: src/stores/signupStore.js (hunk 5 above)
Comment (aayushbatra02) [reply src/stores/signupStore.js]: ok sir

**Pull request merged**